### PR TITLE
Imbu runs NLP models with new token-level scheme

### PR DESCRIPTION
**Pull request by BoltzmannBrain**
* Models are trained by feeding them one token at a time.
* Also removed the CioWindows model for now; it'll likely be replaced with a model that uses NuPIC's Network API.

@oxtopus please review.

### CODE REVIEW

File: imbu/engine/fluent_api.py [hunk 1]
```diff
@@ -332,6 +340,23 @@ def createModel(modelName, modelFactory):
   g_models[modelName] = modelProxy
 
 
+def trainModelWithText(model, trainingData):
+  """ Train the given model on trainingData.
+  This is (essentially) the same training method as in the research repo's
+  imbu_runner.py.
+  """
+  TP = TextPreprocess()
```
Comment (oxtopus) [inline imbu/engine/fluent_api.py]: Per naming convention, the capitalized `TP` implies that it's a class.  Meanwhile, `TP` is commonly used elsewhere to refer to "Temporal Pooler".  Consider naming to something more descriptive, say, `testPreprocessor`.

File: imbu/engine/fluent_api.py [hunk 2]
```diff
@@ -332,6 +340,23 @@ def createModel(modelName, modelFactory):
...
+  for seqId, (text, _, _) in enumerate(trainingData.values()):
+    textTokens = TP.tokenize(text)  # TODO: use model's tokenization method instead
+    lastToken = len(textTokens) - 1
+    for i, token in enumerate(textTokens):
+      # use the sequence's ID as the category label
+      model.trainText(token,
+                      [seqId],
+                      sequenceId=seqId,
+                      reset=int(i==lastToken))
```
Comment (oxtopus) [inline imbu/engine/fluent_api.py]: Why is this param an `int` and not `bool`?

Comment (BoltzmannBrain) [inline imbu/engine/fluent_api.py]: The NLP models expect `int` resets b/c this is how resets are flagged in NuPIC's network API.

File: imbu/engine/fluent_api.py (hunk 1 above)
Comment (BoltzmannBrain) [inline imbu/engine/fluent_api.py]: yes thank you

**Pull request merged**